dashboard/test/lib/quick_assign_helper.rb

```diff
@@ -38,25 +56,47 @@ class QuickAssignHelperTest < ActionController::TestCase
     course_offering4 = create :course_offering, curriculum_type: 'Standalone Unit', header: 'Header 2', display_name: 'Z'
     course_offering5 = create :course_offering, curriculum_type: 'Standalone Unit', header: 'Header 2', display_name: 'B'
 
     teacher = create :teacher
-    grouped_offerings = QuickAssignHelper.group_offerings([course_offering1, course_offering2, course_offering3, course_offering4, course_offering5], teacher, I18n.default_locale)
+    grouped_offerings = QuickAssignHelper.group_grade_level_offerings([course_offering1, course_offering2, course_offering3, course_offering4, course_offering5], teacher, I18n.default_locale)
 
-    assert_equal 1, grouped_offerings['Course']['Header 1'].length
-    assert_equal course_offering1.id, grouped_offerings['Course']['Header 1'][0][:id]
     assert_equal 1, grouped_offerings['Course']['Header 2'].length
+    assert_equal course_offering1.id, grouped_offerings['Course']['Header 1'][0][:id]
+    assert_equal 1, grouped_offerings['Course']['Header 1'].length
     assert_equal course_offering2.id, grouped_offerings['Course']['Header 2'][0][:id]
     assert_equal 1, grouped_offerings['Standalone Unit']['Header 1'].length
     assert_equal course_offering3.id, grouped_offerings['Standalone Unit']['Header 1'][0][:id]
     assert_equal 2, grouped_offerings['Standalone Unit']['Header 2'].length
     assert_equal course_offering5.id, grouped_offerings['Standalone Unit']['Header 2'][0][:id]
     assert_equal course_offering4.id, grouped_offerings['Standalone Unit']['Header 2'][1][:id]
+
+    assert_equal ['Header 1', 'Header 2'], grouped_offerings['Course'].keys
+  end
+
+  test 'HoC course offerings are grouped by header' do
+    course_offering1 = create :course_offering, marketing_initiative: 'HOC', header: 'A Header'
+    course_offering2 = create :course_offering, marketing_initiative: 'HOC', header: 'B Header', display_name: 'Z'
+    course_offering3 = create :course_offering, marketing_initiative: 'HOC', header: 'B Header', display_name: 'A'
+    favorite_course_offering = create :course_offering, marketing_initiative: 'HOC', header: 'Favorites'
+
+    teacher = create :teacher
+    grouped_offerings = QuickAssignHelper.group_hoc_and_pl_offerings([course_offering1, course_offering2, course_offering3, favorite_course_offering], teacher, I18n.default_locale)
+
+    assert_equal 1, grouped_offerings['A Header'].length
+    assert_equal course_offering1.id, grouped_offerings['A Header'][0][:id]
+    assert_equal 2, grouped_offerings['B Header'].length
+    assert_equal course_offering3.id, grouped_offerings['B Header'][0][:id]
+    assert_equal course_offering2.id, grouped_offerings['B Header'][1][:id]
+    assert_equal 1, grouped_offerings['Favorites'].length
+    assert_equal favorite_course_offering.id, grouped_offerings['Favorites'][0][:id]
+
+    assert_equal ['Favorites', 'A Header', 'B Header'], grouped_offerings.keys
```

Comment:
Thank you for testing this!

Reply:
I'm glad I did because it was way harder than I expected 🙃